compiler-rt/test/profile/instrprof-api.c

```diff
@@ -0,0 +1,38 @@
+// RUN: %clang_profgen %s --target=ppc64le-unknown-linux-gnu -S \
+// RUN:    -emit-llvm -o - | FileCheck %s --check-prefix=PROFGEN
+// RUN: %clang_profgen -o %t %s
+// RUN: env LLVM_PROFILE_FILE=%t.profraw %run %t
```

Comment:
I think you need a `REQUIRES: ppc` for this test otherwise it will fail when this line is executed on non-PPC build bots?

Reply:
After reading your comment, I was about to say the binary is compiled with `ppc64le` and requires `host-byteorder-little-endian` but realized the `--target=ppc64le-unknown-linux-gnu` is only used with `-emit-llvm`, and the `RUN` lines that build executables doesn't specify a target.

Nevertheless, I wonder test coverage for `%clang_pgogen` should be added here as well.

Relatedly, I do get compilation errors on ppc big-endian systems for `clang_pgogen` in another compiler-rt test ([commit](https://github.com/llvm/llvm-project/commit/5136c167a2573fbd05179849cb41b198c4862b12) and [buildbot link](https://lab.llvm.org/buildbot/#/builders/18/builds/13228)

From the error message, I *think* lack of ABI implementation should manifest whether it's `clang_pgogen` or `clang_profgen`.

Reply:
Good point, can we drop the `--target=ppc64le-unknown-linux-gnu` flag? It shouldn't make a difference to what we want to test here.

Reply:
> Relatedly, I do get compilation errors on ppc big-endian systems for clang_pgogen in another compiler-rt test ([commit](https://github.com/llvm/llvm-project/commit/5136c167a2573fbd05179849cb41b198c4862b12) and [buildbot link](https://lab.llvm.org/buildbot/#/builders/18/builds/13228))

> From the error message, I think lack of ABI implementation should manifest whether it's clang_pgogen or clang_profgen.

Hmmm thanks for pointing this out! I don't have access to a BE machine at the moment to reproduce but I will go find one and take a look.

Reply:
> I don't have access to a BE machine at the moment to reproduce but I will go find one and take a look.

Or maybe just add `REQUIRES: host-byteorder-little-endian` if this test passes on little-endian systems.

Reply:
Ah thanks for the proposal! 

We still want to keep this test working for both BE and LE because we want this test to run on AIX as well. In its current shape, the test passes on AIX. If it is fine, given [this comment](https://github.com/llvm/llvm-project/pull/76471#issuecomment-1874671580), I can be a bit speculative and let the buildbot alert me if this test is failing on Linux ppc BE.